tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/ValidateAMQPErrorsTest.java

```diff
@@ -0,0 +1,643 @@
+               serverReceiver.setOfferedCapabilities(new Symbol[]{AMQPMirrorControllerSource.MIRROR_CAPABILITY});
+
+               serverReceiver.handler((del, msg) -> {
+                  logger.debug("prefetch = " + serverReceiver.getPrefetch() + ", Got message: " + msg);
+                  if (msg.getApplicationProperties() != null) {
+                     Map map = msg.getApplicationProperties().getValue();
+                     Object value = map.get("sender");
+                     if (value != null) {
+                        if (messagesReceived.get() != ((Integer) value).intValue()) {
+                           logger.warn("Message out of order. Expected " + messagesReceived.get() + " but received " + value);
+                           errors.incrementAndGet();
+                        }
+                        messagesReceived.incrementAndGet();
+                     }
+                  }
+                  del.disposition(Accepted.getInstance(), true);
+                  if (serverReceiver.getPrefetch() == 0) {
+                     serverReceiver.flow(1);
+                  }
+               });
+
+               serverReceiver.open();
+            }
+         });
+      });
+
+      AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration("test", "tcp://localhost:" + mockServer.actualPort()).setReconnectAttempts(3).setRetryInterval(10);
+      amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement());
+      server.getConfiguration().addAMQPConnection(amqpConnection);
+      server.start();
+
+      Wait.assertEquals(1, disconnected::size, 6000);
+      Wait.assertEquals(2, connections::size, 6000);
+
+      assertSame(connections.get(0), disconnected.get(0));
+      assertFalse(connections.get(1).isDisconnected());
+
+      assertEquals("Should not have got any message on refused link", 0, refusedLinkMessageCount.get());
+      assertEquals(0, errors.get());
+
+      ConnectionFactory factory = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:" + AMQP_PORT);
+      try (Connection connection = factory.createConnection()) {
+         Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+         MessageProducer producer = session.createProducer(session.createQueue(getQueueName()));
+         for (int i = 0; i < 100; i++) {
+            TextMessage message = session.createTextMessage("hello");
+            message.setIntProperty("sender", i);
+            producer.send(message);
+         }
+      }
+
+      Wait.assertEquals(100, messagesReceived::intValue);
+      assertEquals(0, errors.get()); // Meant to check again. the errors check before was because of connection issues. This one is about duplicates on receiving
+   }
+
+   /**
+    * Refuse the first mirror link, verify broker handles it and reconnects
+    *
+    * @throws Exception
+    */
+   @Test
+   public void testClientMissingCapabilities() throws Exception {
```

Comment:
```suggestion
   public void testNoClientDesiredMirrorCapability() throws Exception {
```
For better symmetry with the other existing test. Could update them both to align in another way.